packages/cli/src/UserManagement/email/UserManagementMailer.ts

```diff
@@ -12,35 +13,33 @@ import {
 } from './Interfaces';
 import { NodeMailer } from './NodeMailer';
 
-// TODO: make function fully async (remove sync functions)
-async function getTemplate(configKeyName: string, defaultFilename: string) {
-	const templateOverride = (await GenericHelpers.getConfigValue(
-		`userManagement.emails.templates.${configKeyName}`,
-	)) as string;
+type Template = HandlebarsTemplateDelegate<unknown>;
+type TemplateName = 'invite' | 'passwordReset';
 
-	let template;
-	if (templateOverride && existsSync(templateOverride)) {
-		template = readFileSync(templateOverride, {
-			encoding: 'utf-8',
-		});
-	} else {
-		template = readFileSync(pathJoin(__dirname, `templates/${defaultFilename}`), {
-			encoding: 'utf-8',
-		});
+const templates: Partial<Record<TemplateName, Template>> = {};
+
+async function getTemplate(
+	templateName: TemplateName,
+	defaultFilename = `${templateName}.html`,
+): Promise<Template> {
+	let template = templates[templateName];
+	if (!template) {
+		const templateOverride = (await GenericHelpers.getConfigValue(
+			`userManagement.emails.templates.${templateName}`,
+		)) as string;
+
+		let markup;
+		if (templateOverride && existsSync(templateOverride)) {
+			markup = await readFile(templateOverride, 'utf-8');
+		} else {
+			markup = await readFile(pathJoin(__dirname, `templates/${defaultFilename}`), 'utf-8');
+		}
+		template = Handlebars.compile(markup);
+		templates[templateName] = template;
 	}
 	return template;
 }
 
-function replaceStrings(template: string, data: IDataObject) {
-	let output = template;
-	const keys = Object.keys(data);
-	keys.forEach((key) => {
-		const regex = new RegExp(`\\{\\{\\s*${key}\\s*\\}\\}`, 'g');
```

Comment:
when we use regexp-replace for templating, it creates an additional burdun of sanitizing all inputs that go into the template (to avoid things like XSS).
Handlebars takes care of that.